src/rt/test/func/memory/memory_gc.h

```diff
@@ -426,47 +426,97 @@ namespace memory_gc
     alloc_in_region<Cx, Fx, Cx, Fx>(alloc, o); // unreachable
 
     // Run a GC.
-    assert(Region::debug_size(o) == 13);
+    check(Region::debug_size(o) == 13);
     RegionTrace::gc(alloc, o);
-    assert(Region::debug_size(o) == 9);
+    check(Region::debug_size(o) == 9);
 
     // Swap root, but this creates garbage.
     RegionTrace::swap_root(o, nroot);
-    assert(!o->debug_is_iso() && nroot->debug_is_iso());
+    check(!o->debug_is_iso() && nroot->debug_is_iso());
 
     // Run another GC.
     RegionTrace::gc(alloc, nroot);
-    assert(Region::debug_size(nroot) == 4);
+    check(Region::debug_size(nroot) == 4);
 
     // Create another region.
     o = new (alloc) Cx;
     o->f1 = new (alloc, o) Fx;
     o->f1->c1 = new (alloc, o) Cx;
     auto* nnroot = new (alloc, o) Cx;
     o->f1->c2 = nnroot;
-    assert(Region::debug_size(o) == 4);
+    check(Region::debug_size(o) == 4);
 
     // Merge the regions.
     RegionTrace::merge(alloc, nroot, o);
     nroot->c2 = o;
-    assert(Region::debug_size(nroot) == 8);
+    check(Region::debug_size(nroot) == 8);
 
     // Swap root again.
     RegionTrace::swap_root(nroot, nnroot);
-    assert(!nroot->debug_is_iso() && nnroot->debug_is_iso());
+    check(!nroot->debug_is_iso() && nnroot->debug_is_iso());
 
     // Run another GC.
-    assert(Region::debug_size(nnroot) == 8);
+    check(Region::debug_size(nnroot) == 8);
     RegionTrace::gc(alloc, nnroot);
-    assert(Region::debug_size(nnroot) == 1);
+    check(Region::debug_size(nnroot) == 1);
 
     Region::release(alloc, nnroot);
     snmalloc::current_alloc_pool()->debug_check_empty();
   }
 
+  void test_additional_roots()
```

Comment:
Could you extend this test / add another one that checks that objects reachable from an additional root (but which aren't roots themselves) are preserved. It would make sure we not only mark the roots, but we trace them too.